cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -955,8 +957,8 @@ public void liveInfo(HttpRequest request, HttpResponder responder, @PathParam("n
                        @PathParam("program-id") String programId) {
     ProgramType type = getProgramType(programCategory);
     if (type == null) {
-      responder.sendString(HttpResponseStatus.METHOD_NOT_ALLOWED, String.format("Live-info not supported for program" +
-                                                                                  " type '%s'", programCategory));
+      responder.sendString(HttpResponseStatus.BAD_REQUEST,
```

Comment:
But actually this isn't really in the scope of this PR, so I'll revert that change.